python/delta/exceptions.py

```diff
@@ -119,8 +124,20 @@ def _convert_delta_exception(e: "JavaObject") -> Optional[CapturedException]:
     c: "JavaObject" = e.getCause()
 
     jvm: "JVMView" = SparkContext._jvm  # type: ignore[attr-defined]
+    gw = SparkContext._gateway  # type: ignore[attr-defined]
     stacktrace = jvm.org.apache.spark.util.Utils.exceptionString(e)
 
+    # Temporary workaround until Delta Lake is upgraded to Spark 3.3
+    # Below three exception handling cases are copied from
+    # https://github.com/apache/spark/blob/master/python/pyspark/sql/utils.py#L156
+    if is_instance_of(gw, e, "org.apache.spark.sql.catalyst.parser.ParseException"):
```

Comment:
any test cases for this @vkorukanti ?

Reply:
We already have one existing test [here](https://github.com/delta-io/delta/blob/master/python/delta/tests/test_deltatable.py#L716).